schemas/participant.ts

```diff
@@ -3,18 +3,49 @@ import { z } from 'zod';
 export const participantIdentifierSchema = z
   .string()
   .min(1, { message: 'Identifier cannot be empty' })
-  .max(255, { message: 'Identifier too long. Maxiumum of 255 characters.' });
+  .max(255, { message: 'Identifier too long. Maximum of 255 characters.' })
+  // Ensure that the identifier is not just whitespace or have leading/trailing spaces
+  .refine((identifier) => identifier.trim() === identifier, {
+    message: 'Identifier cannot have leading or trailing spaces',
+  });
 
-export const participantLabelSchema = z.string().optional();
+const participantIdentifierOptionalSchema = z
+  .string()
+  .max(255, {
+    message: 'Identifier too long. Maximum of 255 characters.',
+  })
+  .optional()
+  .refine(
+    (identifier) =>
+      identifier === undefined || identifier.trim() === identifier,
```

Comment:
I don't think the check for `identifier === undefined` is needed here, but if it _is_ (because of `optional()`) it might be better to check if identifier is defined (returning early if it isnt) and then implement a purer version of the "check for only whitespace" code. I think this will make it more readable.

You should consider making the whitespace check a utility function so it can be reused. It could follow the pattern of the other zod utilities.

I also think the message here is misleading as per my comments above. It should be updated the same way, with `.trim()` added at the end.

Reply:
resolved [7141e0c](https://github.com/complexdatacollective/Fresco/pull/140/commits/7141e0c494fae69109e23f0b2b45482981dbae36)